### ISSUE TITLE
Moloch Not writing Data

### ISSUE THREAD

**Issue by grantcurell** [aol/moloch#869]
6.2.4

**Moloch version**:
1.1.1

**OS name and version**:
Red Hat 7.5

**How was Moloch built/installed:** (rpm, deb, singlehost built, ...)
rpm

**Provide logs, stack traces and steps to reproduce**:
Moloch writes nothing to disk. I gave up and gave 777 permissions to the entire /data/moloch/raw folder. Checked to make sure the interface Moloch is set to listen to was receiving data with tcpdump. Nothing in the logs capture or viewer, it's connected to Elastic without issue. The PCAP file appears in the folder, but the size is stuck at 0 and nothing appears in the viewer (though the viewer displays my node as up and green)

I've played all sorts of sessions, browsed the web, everything. Still stuck at 0.

**Comment by awick** [comment]
https://github.com/aol/moloch/wiki/FAQ#moloch-is-not-working
https://github.com/aol/moloch/wiki/FAQ#zero-byte-pcap-files

if those don't help join slack

**Comment by grantcurell** [comment]
For anyone else who comes looking at this, Moloch will not even write data if Elasticsearch is not green. The PCAP size will sit at zero.

In my case, Moloch successfully connected to my Elasticsearch node, but the other two nodes in the cluster were not connected so the health was yellow. This meant that Moloch shows good to go across the board in the logs, but in reality was not.

Suggestion: Logs should reflect that Moloch is unable to write data because of cluster health. Obviously cluster health yellow is not good, but Elasticsearch still ingests data without issue for other applications with Moloch being an exception. It would be helpful if the logs said something about cluster health being an issue.

**Comment by awick** [comment]
Actually the above isn't accurate, Moloch works fine with yellow status, so this isn't different with other tools.  Yellow = "Elasticsearch has allocated all of the primary shards, but some/all of the replicas have not been allocated."  This sounds more like a split brain issue or something else, were the viewer can capture pointing to different es nodes?  If you have exact steps how to reproduce that would be awesome.

As far as stating that ES is yellow, its in the upper right corner.

**Comment by grantcurell** [comment]
No, both were pointed to the same DNS address and the moment I fired up the other two ES nodes and my cluster status turned green, Moloch immediately started working.

**Comment by awick** [comment]
So the only way I was able to reproduce was to
* run all es nodes
* run db.pl init, so moloch/elasticsearch things there are 3 nodes
* shutdown 2 es nodes
* start capture
* elasticsearch cluster immediately turns red, and msg in  elasticsearch log about not being able to create indices
* start 2 nodes, elasticsearch turns back green

When this happens elasticsearch does not immediately respond to capture what is wrong.  Does this match?  I've added periodic printing of elasticsearch status to capture.log, but it is also in the elasticsearch.log and viewer ui already.
